src/sentry/tasks/post_process.py

```diff
@@ -674,7 +675,20 @@ def process_snoozes(job: PostProcessJob) -> None:
         and group.status == GroupStatus.IGNORED
         and group.substatus == GroupSubStatus.UNTIL_ESCALATING
     ):
-        job["has_reappeared"] = is_escalating(group)
+        if is_escalating(group):
+            group.substatus = GroupSubStatus.ESCALATING
+            group.status = GroupStatus.UNRESOLVED
+            group.save(update_fields=["status", "substatus"])
+            add_group_to_inbox(group, GroupInboxReason.ESCALATING)
+            record_group_history(group, GroupHistoryStatus.ESCALATING)
+
+            analytics.record(
+                "issue.escalating",
+                organization_id=group.project.organization.id,
+                project_id=group.project.id,
+                group_id=group.id,
+            )
+            job["has_reappeared"] = True
```

Comment:
nit: Should this be a function in `post_process`? e.g. `update_as_escalating`

Reply:
we have an `escalating.py` in `/issues` that deals with status-related logic. Might be a better home if we break this into a function like `mark_escalating`.